bytecomp/translmod.ml

```diff
@@ -274,25 +268,26 @@ let reorder_rec_bindings bindings =
   let num_bindings = Array.length id in
   let status = Array.make num_bindings Undefined in
   let res = ref [] in
-  let rec emit_binding i =
+  let rec emit_binding parent i =
     match status.(i) with
       Defined -> ()
-    | Inprogress ->
-        let cycle = extract_unsafe_cycle id fv i in
+    | Inprogress _ ->
+        status.(i) <- Inprogress parent;
```

Comment:
I'm not fond of the write here; my intuition would be that you could do without it if you changed the stopping condition in the extraction code: check `i = stop` before looking at `status.(i)` (so with the un-shadowed `i` variable).

Reply:
I would prefer to keep this write here since it closes the dependency cycle and makes the `extract_unsafe_cycle` works  starting from any point in the cycle.

Reply:
But it also means that you erase/lose information on the way (for example, you cannot recover which initial module provoked the cycle detection). But if you like it this way, ok, you are entitled to your (dubious) preference :-)